src/calendar/day.jsx

```diff
@@ -1,53 +1,69 @@
-import React from 'react'
-import PropTypes from 'prop-types'
-
-import classnames from 'classnames'
-
-import {BLOCK_CLASS_NAME} from './consts'
-import {datePropType} from './_lib'
-
+import cx from 'classnames'
 import formatDate from 'date-fns/format'
+import PropTypes from 'prop-types'
+import React from 'react'
 
-export default class Day extends React.Component {
-  static propTypes = {
-    blockClassName: PropTypes.string,
-    className: PropTypes.string,
-    date: datePropType.isRequired,
-    onClick: PropTypes.func,
-    onMouseMove: PropTypes.func,
-    today: datePropType.isRequired,
-  }
-
-  static defaultProps = {
-    blockClassName: BLOCK_CLASS_NAME,
-  }
-
-  _onClick(e) {
-    e.preventDefault()
-    const {date, onClick} = this.props
-    if (onClick) {
-      onClick(date)
-    }
-  }
-
-  _onMouseMove(e) {
-    e.preventDefault()
-    const {date, onMouseMove} = this.props
-    if (onMouseMove) {
-      onMouseMove(date)
-    }
-  }
+const Day = ({
+  blockClassName,
```

Comment:
For a future enhancement:
this approach its easier to apply a render prop approach